tensorflow_io/oss/kernels/ossfs/oss_file_system.cc

```diff
@@ -0,0 +1,1158 @@
+  for (auto path : listing) {
+    if (str_util::StartsWith(StringPiece(path), base_prefix)) {
+      const string file_path = io::JoinPath(dir, path);
+      if (Env::Default()->MatchPath(file_path, pattern)) {
+        results->push_back(dir + "/" + path);
+      }
+    }
+  }
+
+  return Status::OK();
+}
+
+Status OSSFileSystem::_DeleteObjectInternal(oss_request_options_t* options,
+                                            const std::string& bucket,
+                                            const std::string& object) {
+  aos_string_t bucket_;
+  aos_string_t object_;
+  aos_table_t* resp_headers = NULL;
+  aos_status_t* s = NULL;
+
+  aos_str_set(&bucket_, bucket.c_str());
+  aos_str_set(&object_, object.c_str());
+
+  s = oss_delete_object(options, &bucket_, &object_, &resp_headers);
+  if (!aos_status_is_ok(s)) {
+    string msg;
+    oss_error_message(s, &msg);
+    VLOG(1) << "delete " << object << " failed, errMsg: " << msg;
+    return errors::Internal("delete failed: ", object, " errMsg: ", msg);
+  }
+
+  return Status::OK();
+}
+
+Status OSSFileSystem::DeleteFile(const std::string& fname) {
+  TF_RETURN_IF_ERROR(oss_initialize());
+  std::string object, bucket;
+  TF_RETURN_IF_ERROR(_ParseOSSURIPath(fname, &bucket, &object));
+  OSSConnection oss(host_, access_id_, access_key_);
+  oss_request_options_t* oss_options = oss.getRequestOptions();
+
+  return _DeleteObjectInternal(oss_options, bucket, object);
+}
+
+Status OSSFileSystem::CreateDir(const std::string& dirname) {
+  TF_RETURN_IF_ERROR(oss_initialize());
+  std::string object, bucket;
+  TF_RETURN_IF_ERROR(_ParseOSSURIPath(dirname, &bucket, &object));
+  OSSConnection oss(host_, access_id_, access_key_);
+  oss_request_options_t* ossOptions = oss.getRequestOptions();
+  aos_pool_t* pool = oss.getPool();
+  StringPiece dirs(object);
+
+  std::vector<std::string> splitPaths =
+      str_util::Split(dirs, '/', str_util::SkipEmpty());
+  if (splitPaths.size() < 2) {
+    return _CreateDirInternal(pool, ossOptions, bucket, object);
+  }
+
+  FileStatistics stat;
+  StringPiece parent = io::Dirname(dirs);
+  if (!_StatInternal(pool, ossOptions, bucket, string(parent), &stat).ok()) {
+    return errors::Internal("parent does not exists: ", parent);
+  }
+
+  if (!stat.is_directory) {
+    return errors::Internal("can not mkdir because parent is a file: ", parent);
+  }
+
+  TF_RETURN_IF_ERROR(_CreateDirInternal(pool, ossOptions, bucket, object));
+  return Status::OK();
+}
+
+Status OSSFileSystem::RecursivelyCreateDir(const string& dirname) {
+  TF_RETURN_IF_ERROR(oss_initialize());
+  std::string object, bucket;
+  TF_RETURN_IF_ERROR(_ParseOSSURIPath(dirname, &bucket, &object));
+  OSSConnection oss(host_, access_id_, access_key_);
+  oss_request_options_t* ossOptions = oss.getRequestOptions();
+  aos_pool_t* pool = oss.getPool();
+  StringPiece dirs(object);
+
+  std::vector<std::string> splitPaths =
+      str_util::Split(dirs, '/', str_util::SkipEmpty());
```

Comment:
Consider moving this to a separate function given it appears in multiple places

Reply:
There are only two usages,  and extracting a function involving passing multiple parameters and i think it actually hurts the readability?